### DISCUSSION TITLE
Will the ts decorators configuration be used during build?

### DISCUSSION THREAD

**Discussion by vaynevayne** [vitejs/vite-plugin-react-swc#246] [Ideas]
I saw that if a plug-in is used, swc will be used in the production environment. Do tsDecorators count as a plug-in?
   `tsDecorators: true `

**Comment by ArnaudBarre** [comment]
Nope, esbuild will do the decorators transformation during builds.
Note that it's highly discouraged to use TS "experimentalDecorators" that is the legacy spec and is deprecated
Better support for TC39 decorators is tracked here: https://github.com/vitejs/vite-plugin-react-swc/issues/86
